### PR TITLE
CAMEL-14654: Add Camel Azure Storage Blob component based on Azure v12 SDK

### PR DESCRIPTION
**Pull request by omarsmak**
This a brokedown component that is dedicated to Azure Storage blob based on the component `camel-azure`, but it uses the latest Azure SDK v12 which required some refactoring in order facilitate v12 Azure SDK. The functionality is almost the same as `camel-azure` but with some added features like container operations, return enriched headers, improved docs .. etc. As a second stage, we would like as well to add the batch consumer and some operations related to the snapshot. But that after I port the queue component from the legacy camel-azure.
Once I break down the queue component and change it to version v12, we can mark the camel-azure component as `legacy` in order to highlight this to user as it uses the legacy v8 azure sdk.

- [x]  Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by oscerd** [comment]
It looks good me, we need to create a spring boot starter for this and a karaf feature with the azure storage stuff wrapped for the moment

File: components/camel-azure-storage-blob/src/test/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## ---------------------------------------------------------------------------
+appender.file.type = File
+appender.file.name = file
+appender.file.fileName = target/camel-github-test.log
```
Comment (dmvolod) [inline components/camel-azure-storage-blob/src/test/resources/log4j2.properties:19]: @omarsmak please change to the correct one

Comment (omarsmak) [reply components/camel-azure-storage-blob/src/test/resources/log4j2.properties]: Thanks @dmvolod for spotting, fixed it

**Comment by omarsmak** [comment]
Sure @oscerd. Will add the starters for spring boot and karaf feature once this is merged

**Comment by davsclaus** [comment]
LGTM. Squash the PR before merging

**Pull request merged**